libs/SalesforceSDK/src/com/salesforce/androidsdk/auth/SfapApiGenerations.kt

```diff
@@ -0,0 +1,85 @@
+package com.salesforce.androidsdk.auth
+
+import com.salesforce.androidsdk.auth.HttpAccess.DEFAULT
+import com.salesforce.androidsdk.auth.SfapApiGenerationsResponseBody.Companion.fromJson
+import kotlinx.serialization.json.Json.Default.encodeToString
+import okhttp3.MediaType.Companion.toMediaTypeOrNull
+import okhttp3.Request
+import okhttp3.RequestBody.Companion.toRequestBody
+import java.io.IOException
+
+/**
+ * Provides generated text from the `sfap_api` "generations" endpoint.
+ *
+ * See https://developer.salesforce.com/docs/einstein/genai/guide/access-models-api-with-rest.html
+ *
+ * @param apiHostName The Salesforce `sfap_api` hostname
+ * @param modelName The model name to request generations from.  For possible
+ * values, see https://developer.salesforce.com/docs/einstein/genai/guide/api-names.html
+ */
+class SfapApiGenerations(
+    private val apiHostName: String = API_HOST_NAME_PROD,
+    private val modelName: String
+) {
+
+    /**
+     * Fetches generated text from the `sfap_api` "generations" endpoint.
+     * @param prompt The prompt request parameter value
+     * @param accessToken The request's access token
+     * @return The endpoint's response
+     */
+    @Throws(
+        IOException::class,
+        OAuth2InvalidTokenException::class,
+        SfapApiException::class
+    )
+    fun fetch(
+        prompt: String,
+        accessToken: String
```

Comment:
Why don't we use a RestRequest and the RestClient for those calls? That way the application would no need to handle the access token. Also if the token expires, refresh would be handled automatically.

Reply:
@wmathurin - Take a look at https://github.com/forcedotcom/SalesforceMobileSDK-Android/pull/2644/commits/8218fea8fc27c6ae0159366b54c64bc2275fa2be for the updated REST client.